src/item.cpp

```diff
@@ -1870,9 +1871,11 @@ nc_color item::color_in_inventory() const
         switch( rating ) {
             case EDIBLE:
             case TOO_FULL:
-                if( is_going_bad() ) {
+                if( preserves ) {
+                    // Nothing, canned food won't rot
```

Comment:
Won't this prevent cans from showing perishable contents in their correct color?